pkgs/development/interpreters/gnudatalanguage/default.nix

```
@@ -0,0 +1,156 @@
+, enableMPI ? (stdenv.isLinux || stdenv.isDarwin)
+# Choose mpich because of it builds on aarch and darwin
+, libraryMPI ? mpich
+, mpich
+# Unfree optional dependency
+# for hdf4 and hdf5
+, enableSzip ? false, szip
+# wxWidgets is preferred over X11 for this project
+# but we only have it on Linux and Darwin in Nixpkgs
+, enableWX ? (stdenv.isLinux || stdenv.isDarwin)
+, wxGTK31, wxmac
+# X11: OFF by default for platform consistency
+# Use X where WX is not available
+, enableXWin ? (!stdenv.isLinux && !stdenv.isDarwin)
+, xorg
+}:
+
+let
+  wxWidgets = (if stdenv.isDarwin then wxmac else wxGTK31);
+  enablePlplotDrivers = enableWX || enableXWin;
+  hdf4-custom = (hdf4.override (
+    {
+      uselibtirpc = enableLibtirpc;
+    } // (lib.optionalAttrs enableSzip {
+      inherit szip;
+    })
+  ));
+  hdf5-custom = (hdf5.override (
+    (lib.optionalAttrs enableMPI {
+      mpiSupport = true;
+      mpi = libraryMPI;
+    }) // (lib.optionalAttrs enableSzip {
+      inherit szip;
+    })
+  ));
+  plplot-with-drivers = (plplot.overrideAttrs (oldAttrs: {
+    buildInputs = (oldAttrs.buildInputs or []
+      ++ lib.optional enableWX wxWidgets
+      ++ lib.optional enableXWin xorg.libX11
+    );
+  }));
+in stdenv.mkDerivation rec {
+  pname = "gnudatalanguage";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = "gdl";
+    rev = "v${version}";
+    sha256 = "sha256-Y9LVRaWjQqpWqjNngxB406PE/rl/9S8rs0u0CK5ivUA=";
+  };
+
+  buildInputs = [
+    readline
+    ncurses
+    zlib
+    gsl
+    openmp
+    graphicsmagick
+    netcdf
+    fftw fftwFloat fftwLongDouble
+    proj
+    shapelib
+    expat
+    libraryMPI
+    udunits
+    eigen
+    pslib
+    eccodes
+    glpk
+    libpng
+    libtiff
+    libgeotiff
+    libjpeg
+    hdf4-custom hdf5-custom
+    plplot-with-drivers
+  ] ++ lib.optional enableXWin xorg.libX11
+  ++ lib.optional enableWX wxWidgets
+  ++ lib.optional enableMPI libraryMPI
+  ++ lib.optional enableLibtirpc libtirpc
+  ++ lib.optional enableSzip szip;
+
+  propagatedBuildInputs = [
+    (pythonInterpreter.withPackages (ps: with ps; [ numpy ]))
+  ];
+
+  nativeBuildInputs = [
+    cmake
+  ] ++ lib.optional enableWX wrapGAppsHook;
+
+  cmakeFlags =
+  lib.optional (!enablePlplotDrivers) "-DINTERACTIVE_GRAPHICS=OFF"
+  ++ lib.optional (!enableWX) "-DWXWIDGETS=OFF"
+  ++ lib.optional enableSzip "-DSZIPDIR=${szip}"
+  ++ lib.optionals enableXWin [ "-DX11=ON" "-DX11DIR=${xorg.libX11}" ]
+  ++ lib.optionals enableMPI [ "-DMPI=ON" "-DMPIDIR=${libraryMPI}" ];
+
+  doCheck = true;
+
+  # Opt-out unstable tests
+  # https://github.com/gnudatalanguage/gdl/issues/482
+  # https://github.com/gnudatalanguage/gdl/issues/1079
+  # https://github.com/gnudatalanguage/gdl/issues/460
+  preCheck = ''
+    checkFlagsArray+=("ARGS=-E 'test_tic_toc.pro|test_byte_conversion.pro|test_bytscl.pro|test_call_external.pro'")
+  '';
+
+  passthru = {
+     inherit  hdf4-custom hdf5-custom plplot-with-drivers;
```

I'd use:

```suggestion
     hdf4 = hdf4-custom;
     hdf5 = hdf5-custom;
     plplot = plplot-with-drivers;
```

This way derivations that depend on this package, will be able to `inherit (gnudatalanguage) hdf5 hdf4 plplot;`.